### PR TITLE
Tab bar elevation

### PR DESCRIPTION
**Pull request by Xavjer**
As a TabBar won't always be used at the bottom of the screen, it may be required that it has an elevation to appear over other content.

We use the existing elevation by wrapping the tabBar within a Material Widget

### CODE REVIEW

**Comment by Xavjer** [comment]
@zoechi Is there a way I can support that this pull request is reviewed? 

With elevation:
withelevationproperty
Without elevation:
withoutelevationproperty

**Comment by hyperpanthera** [comment]
Merge this please.

**Comment by Xavjer** [comment]
Conflicts were resolved and the pull request would be ready to be merged

**Comment by jmagman** [comment]
@HansMuller what would you like to do with this one?

**Comment by hyperpanthera** [comment]
MERGE IT.

File: packages/flutter/lib/src/material/tabs.dart [hunk 1]
```diff
@@ -1100,7 +1108,15 @@ class _TabBarState extends State<TabBar> {
       );
     }
 
-    return tabBar;
+    // Wrap in Material if the tabBar has an elevation
+    if (widget.elevation > 0) {
```
Comment (dnfield) [inline packages/flutter/lib/src/material/tabs.dart]: This needs a test.  It also is now failing analysis and needs to be brought up to date.  Can you update it?

**Comment by dnfield** [comment]
We can't merge this as is because it's failing CI and needs tests.  

If @xavjer can update it that would be great. Otherwise if this is a feature that's needed we should open a new PR with the updated changes and tests.

**Comment by Xavjer** [comment]
@dnfield Done

**Comment by dnfield** [comment]
It still needs a test

**Comment by dnfield** [comment]
This is still failing to analyze properly (probably because it's not merged up to head), and missing tests.

Please feel free to reopen this once those issues are addressed.

**Pull request closed without merging**